### ISSUE TITLE
Don't compile anymore with dmd 2.068

### ISSUE THREAD

**Issue by BBasile** [kiith-sa/D-YAML#35]
```
compiling C:\...\projects\D-YAML.coedit
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(277): Error: safe function 'dyaml.node.Node.__ctor!string.this' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!string.this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(320): Error: template instance dyaml.node.Node.__ctor!string error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\composer.d(242):        instantiated from here: node!(string, immutable(ScalarStyle))
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(332): Error: safe function 'dyaml.node.Node.__ctor!(Node).this' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(Node[]).this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(320): Error: template instance dyaml.node.Node.__ctor!(Node) error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\composer.d(269):        instantiated from here: node!(Node[], immutable(CollectionStyle))
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1756): Error: safe function 'dyaml.node.Node.findPair!(string, cast(Flag)true).findPair' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.get!(const(Pair[])).get'
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(783): Error: template instance dyaml.node.Node.findPair!(string, cast(Flag)true) error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(620):        instantiated from here: opIndex!string
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1098):        instantiated from here: get!(Pair[], cast(Flag)true)
C:\Dev\metad\repos\D-YAML\source\dyaml\composer.d(310):        instantiated from here: opApply!(Node, Node)
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1042): Error: template instance dyaml.node.Node.get!(Node[], cast(Flag)true) error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\composer.d(330):        instantiated from here: opApply!(Node)
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(332): Error: safe function 'dyaml.node.Node.__ctor!(Pair).this' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(Pair[]).this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(320): Error: template instance dyaml.node.Node.__ctor!(Pair) error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\composer.d(387):        instantiated from here: node!(Pair[], immutable(CollectionStyle))
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1431): Error: safe function 'dyaml.node.Node.value!(YAMLNull).value' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(YAMLNull).this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(361): Error: template instance dyaml.node.Node.value!(YAMLNull) error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(189):        instantiated from here: addConstructor!(YAMLNull)
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(88):        instantiated from here: addConstructorScalar!(YAMLNull)
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1431): Error: safe function 'dyaml.node.Node.value!bool.value' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!bool.this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(361): Error: template instance dyaml.node.Node.value!bool error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(189):        instantiated from here: addConstructor!bool
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(89):        instantiated from here: addConstructorScalar!bool
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1431): Error: safe function 'dyaml.node.Node.value!long.value' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!long.this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(361): Error: template instance dyaml.node.Node.value!long error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(189):        instantiated from here: addConstructor!long
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(90):        instantiated from here: addConstructorScalar!long
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1431): Error: safe function 'dyaml.node.Node.value!real.value' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!real.this'
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(361): Error: template instance dyaml.node.Node.value!real error instantiating
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(189):        instantiated from here: addConstructor!real
C:\Dev\metad\repos\D-YAML\source\dyaml\constructor.d(91):        instantiated from here: addConstructorScalar!real
C:\Dev\metad\repos\D-YAML\source\dyaml\node.d(1431): Error: safe function 'dyaml.node.Node.value!(ubyte[]).value' cannot call system function 'std.variant.VariantN!(16u, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(ubyte[]).this'
C:\...\projects\D-YAML.coedit has not been compiled
```

**Comment by btbytes** [comment]
FWIW ... this is the error message - 

```
Target tinyendian 0.1.2 is up to date. Use --force to rebuild.
Building dyaml 0.5.1 configuration "library", build type debug.
Running dmd...
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(277): Error: safe function 'dyaml.node.Node.__ctor!string.this' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!string.this'
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(320): Error: template instance dyaml.node.Node.__ctor!string error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/composer.d(242):        instantiated from here: node!(string, immutable(ScalarStyle))
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(332): Error: safe function 'dyaml.node.Node.__ctor!(Node).this' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(Node[]).this'
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(320): Error: template instance dyaml.node.Node.__ctor!(Node) error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/composer.d(269):        instantiated from here: node!(Node[], immutable(CollectionStyle))
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(1756): Error: safe function 'dyaml.node.Node.findPair!(string, cast(Flag)true).findPair' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.get!(const(Pair[])).get'
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(783): Error: template instance dyaml.node.Node.findPair!(string, cast(Flag)true) error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(620):        instantiated from here: opIndex!string
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(1098):        instantiated from here: get!(Pair[], cast(Flag)true)
../../.dub/packages/dyaml-0.5.1/source/dyaml/composer.d(310):        instantiated from here: opApply!(Node, Node)
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(1042): Error: template instance dyaml.node.Node.get!(Node[], cast(Flag)true) error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/composer.d(330):        instantiated from here: opApply!(Node)
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(332): Error: safe function 'dyaml.node.Node.__ctor!(Pair).this' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(Pair[]).this'
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(320): Error: template instance dyaml.node.Node.__ctor!(Pair) error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/composer.d(387):        instantiated from here: node!(Pair[], immutable(CollectionStyle))
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(1431): Error: safe function 'dyaml.node.Node.value!(YAMLNull).value' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool, long, real, ubyte[], SysTime, string, Pair[], Node[], YAMLObject).VariantN.__ctor!(YAMLNull).this'
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(361): Error: template instance dyaml.node.Node.value!(YAMLNull) error instantiating
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(189):        instantiated from here: addConstructor!(YAMLNull)
../../.dub/packages/dyaml-0.5.1/source/dyaml/constructor.d(88):        instantiated from here: addConstructorScalar!(YAMLNull)
../../.dub/packages/dyaml-0.5.1/source/dyaml/node.d(1431): Error: safe function 'dyaml.node.Node.value!bool.value' cannot call system function 'std.variant.VariantN!(16LU, YAMLNull, YAMLMerge, bool
```
(snipped)

**Comment by kiith-sa** [comment]
Will look at this over the next weekend, probably will need to remove some @safe from the API; I guess less of Variant is @safe due to better attribute inference?

**Comment by kiith-sa** [comment]
Should be fixed by 2f3d782c847fde6917a6faedc90232a7560245fc , can you test?

**Comment by BBasile** [comment]
Thx. Ok now. (verified on win 32 but the issue was platform-agnostic so i dare closing)
